### PR TITLE
Added opengraph meta tags meant for Discord embeds

### PR DESCRIPTION
**Pull request by jacany**
This also removes a return from `show` in MediaController that would give a raw image
This also has to be translated

### CODE REVIEW

**Comment by jacany** [comment]
Preview of the embeds
image

File: resources/templates/base.twig [hunk 1]
```diff
@@ -34,6 +33,23 @@
...
     </script>
+    <meta id="embed-title" property="og:site_name" content="{{ config.app_name }}">
+    <meta id="embed-url" property="og:url" content="{{ url }}">
+    <!-- This will only display an image, but that's the most important for now -->
+    {% if type %}
+        <meta name="twitter:card" content="summary_large_image">
+        <meta property="og:type" content="website" />
+        <meta id="embed-title" property="og:title" content="{{ media.filename }} ({{ media.size }})">
+        <meta id="embed-image" property="og:image" content="{{ url }}/raw">
+        <meta id="embed-desc" property="og:description" content="Uploaded at {{ media.timestamp }} by {{ media.username }}">
```
Comment (jacany) [inline resources/templates/base.twig:44]: This needs to be translatable.

File: app/Controllers/MediaController.php [hunk 2]
```diff
@@ -41,10 +41,6 @@ public function show(Request $request, Response $response, string $userCode, str
 
         $filesystem = $this->storage;
 
-        if (isBot($request->getHeaderLine('User-Agent'))) {
-            return $this->streamMedia($request, $response, $filesystem, $media);
-        }
-
```
Comment (SergiX44) [inline app/Controllers/MediaController.php:47]: You can't simply drop this, will make it work for discord, but for other crawlers like telegram/slack etc may stop working.

File: resources/templates/base.twig [hunk 3]
```diff
@@ -34,6 +33,23 @@
...
+        <meta id="discord" name="twitter:image" content="{{ url }}/raw">
+        <meta id="image-src" name="twitter:image:src" content="{{ url }}/raw">
+    {% else %}
+        <meta id="embed-title" property="og:title" content="{{ block('title') }}">
+        <meta id="discord" name="twitter:image" content="/static/images/favicon-32x32.png">
+        <meta id="image-src" name="twitter:image:src" content="/static/images/favicon-32x32.png">
+        <meta id="embed-image" property="og:image" content="/static/images/favicon-32x32.png">
+    {% endif %}
```
Comment (SergiX44) [inline resources/templates/base.twig:52]: The base template will be always included in each page, so I don't like put some specific code only that works for the public template. I think you should defining it like a block with a default content and the exdending it in the other views

**Review by SergiX44 — CHANGES_REQUESTED** [review]

File: resources/templates/base.twig (hunk 1 above)
Comment (SergiX44) [reply resources/templates/base.twig:44]: As description I would keep only the timestamp, and maybe the list of the tags. I wouldn't expose the internal user name.

File: app/Controllers/MediaController.php (hunk 2 above)
Comment (jacany) [reply app/Controllers/MediaController.php:47]: Thanks, I didn't know that. I will try to make it so it's specifically skips Discord's user agent, i have to find the name of it first though.

File: resources/templates/base.twig (hunk 3 above)
Comment (jacany) [reply resources/templates/base.twig:52]: The point of the if else is to see if there is a type, if not we just assume it's not a link to content

**Comment by josevazquz** [comment]
This happens whenever I try and embed the image. (I copied the 3 files exactly. Not sure If I'm just being stuipd.)
Discord_d1eMYjDyGd
If I add this to custom HTML head content 
```
        <meta property="og:type" content="website" />
        <meta id="embed-title" property="og:title" content="{{ media.filename }} ({{ media.size }})">
        <meta id="embed-image" property="og:image" content="{{ url }}/raw">
        <meta id="embed-desc" property="og:description" content="Uploaded at {{ media.timestamp }} by {{ media.username }}">

```
I get this
Discord_h0hZTzcXQj

Not getting this D:
109983806-2c68b200-7cd1-11eb-9d39-9db9a27946ca

**Comment by jacany** [comment]
> This happens whenever I try and embed the image. (I copied the 3 files exactly. Not sure If I'm just being stuipd.)
> Discord_d1eMYjDyGd
> If I add this to custom HTML head content
> 
> ```
>         <meta property="og:type" content="website" />
>         <meta id="embed-title" property="og:title" content="{{ media.filename }} ({{ media.size }})">
>         <meta id="embed-image" property="og:image" content="{{ url }}/raw">
>         <meta id="embed-desc" property="og:description" content="Uploaded at {{ media.timestamp }} by {{ media.username }}">
> ```
> 
> I get this
> Discord_h0hZTzcXQj
> 
> Not getting this D:
> 109983806-2c68b200-7cd1-11eb-9d39-9db9a27946ca

Youre not supposed to be adding anything into the custom head content field. Also ensure you clear the `resources/cache` folder, and use a different URL with the same image because discord also likes to do some weird caching. Im assuming that's what's going on here because that's not what the code in this pull should be producing.

**Comment by josevazquz** [comment]
> > This happens whenever I try and embed the image. (I copied the 3 files exactly. Not sure If I'm just being stuipd.)
> > Discord_d1eMYjDyGd
> > If I add this to custom HTML head content
> > ```
> >         <meta property="og:type" content="website" />
> >         <meta id="embed-title" property="og:title" content="{{ media.filename }} ({{ media.size }})">
> >         <meta id="embed-image" property="og:image" content="{{ url }}/raw">
> >         <meta id="embed-desc" property="og:description" content="Uploaded at {{ media.timestamp }} by {{ media.username }}">
> > ```
> > 
> > 
> > I get this
> > Discord_h0hZTzcXQj
> > Not getting this D:
> > 109983806-2c68b200-7cd1-11eb-9d39-9db9a27946ca
> 
> Youre not supposed to be adding anything into the custom head content field. Also ensure you clear the `resources/cache` folder, and use a different URL with the same image because discord also likes to do some weird caching. Im assuming that's what's going on here because that's not what the code in this pull should be producing.

It works! Thanks, it was just me being stupid lol. 
Discord_6Ms9Q7qzIZ

**Comment by SergiX44** [comment]
I've added this meta tags with a different implementation here: https://github.com/SergiX44/XBackBone/commit/014b5428aac3b60b7565c6c52e1fad0d2f6b0859

I'm closing this since should not needed anymore.

**Pull request closed without merging**